src/include/duckdb/function/aggregate_function.hpp

```diff
@@ -40,10 +48,15 @@ typedef void (*aggregate_destructor_t)(Vector &state, AggregateInputData &aggr_i
 typedef void (*aggregate_simple_update_t)(Vector inputs[], AggregateInputData &aggr_input_data, idx_t input_count,
                                           data_ptr_t state, idx_t count);
 
-//! The type used for updating complex windowed aggregate functions (optional)
+//! The type used for computing complex/custom windowed aggregate functions (optional)
 typedef void (*aggregate_window_t)(Vector inputs[], const ValidityMask &filter_mask,
                                    AggregateInputData &aggr_input_data, idx_t input_count, data_ptr_t state,
-                                   const vector<FrameBounds> &frames, Vector &result, idx_t rid);
+                                   const SubFrames &frames, Vector &result, idx_t rid, const_data_ptr_t win_state);
```

Comment:
Maybe we should be more explicit? `data_ptr_t` is also a window state (`win_state`). What about `l_state` and `g_state`? I think that this convention is somewhat common for other parts of duckdb code.

Reply:
Good call - I started doing that in other places but never went back to the headers.